include/CFL/CFLGrammar.h

```diff
@@ -51,68 +54,181 @@ class GrammarBase
         return prod.at(pos);
     }
 
+    inline Set<Symbol>& getAttrSyms() 
+    {
+        return this->attributeSymbol;
+    }
+
     Symbol insertTerminalSymbol(std::string strLit);
     Symbol insertNonTerminalSymbol(std::string strLit);
+    void insertAttribute(Symbol s);
+
+    inline static Symbol getNormalKind(Symbol attributedKind)
+    {
+        return (EdgeKindMask & attributedKind);
+    }
+
+    inline static Symbol getAttribute(Symbol attributedKind)
+    {
+        return (attributedKind >> EdgeKindMaskBits);
+    }
+
+    inline static Symbol getAttributedKind(Symbol attribute, Symbol flag)
```

Comment:
We will need to use different types/names for Symbol and Kind and Attribute.
Typedef Kind and Attribute would be better.

Reply:
use GEdgeKind for EdgeKind

Reply:
GEdgeFlag for Attribute